### ISSUE TITLE
VndErrors logref is not just Integer

### ISSUE THREAD

**Issue by jvalkeal** [spring-projects/spring-hateoas#1291]
Looks like constructor is trying to parse `Integer` out from `String`:
https://github.com/spring-projects/spring-hateoas/blob/5d1136ee121fa099ca0ae8fe08572a2b5f10b165/src/main/java/org/springframework/hateoas/mediatype/vnderrors/VndErrors.java#L95

This then for example in a dataflow errors out when trying to upgrade to boot 2.3.x:
```
java.lang.NumberFormatException: For input string: "NoSuchAppRegistrationException"
	at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.lang.Integer.parseInt(Integer.java:580)
	at java.lang.Integer.parseInt(Integer.java:615)
	at org.springframework.hateoas.mediatype.vnderrors.VndErrors.<init>(VndErrors.java:95)
	at org.springframework.cloud.dataflow.server.controller.RestControllerAdvice.onNotFoundException(RestControllerAdvice.java:163)
```
At least https://github.com/blongden/vnd.error mentions `logref` being `numeric/alpha/alphanumeric`

**Comment by gregturn** [comment]
@odrotbohm @jvalkeal I've coded a patch against `master` branch for 1.2. We probably have to backport it to 1.1, 1.0. and 0.25.

**Comment by odrotbohm** [comment]
After investigating this a bit more in detail, it looks like the change made for #775 caused the constructor argument incompatibly change from `String` to `Integer`. That change was released in 1.1 M1, and Spring Boot 2.3 eventually. So everyone using this API trying to upgrade to Spring Boot 2.3 is going to run into this issue. I'll consult with the Boot team about the fix to make this work again causing a breaking API change. If they don't object, I'd back-port this to 1.1.x for inclusion in the next Boot 2.3 as well.
